Sources/Basic/OutputByteStream.swift

```diff
@@ -59,13 +59,13 @@ public class OutputByteStream: OutputStream {
     ///
     /// This method implicitly flushes the stream.
     public var bytes: ByteString {
-        flush()
+        try? flush()
```

Comment:
I think the `bytes` accessor needs to move out of the basic OutputByteStream implementation, and into an implementation which is specifically in-memory.

The default file-based implementation won't be able to honor this call efficiently.

We also need this to become more complicated. For performance reasons, what we want is that the basic OutputByteStream writes its data into a temporary internal buffer (with the expectation that most implementations, e.g., writing a character, are usually just populate this buffer), and that subclasses implement some other method which flushes that buffer to their concrete stream.

Bonus points if the in-memory version avoids doing unnecessary copying, although that is tricky and probably best done in a later iteration.